docs/shared/nx-plugin.md

```diff
@@ -0,0 +1,248 @@
+# Creating a Nx Plugin
+
```

Comment:
This could use an introduction of what an Nx Plugin is, and tell you that the guide shows you how to generate, test, and use an Nx Plugin.